base/base/wide_integer_impl.h

```diff
@@ -1239,7 +1239,7 @@ constexpr integer<Bits, Signed>::operator long double() const noexcept
     for (unsigned i = 0; i < _impl::item_count; ++i)
     {
         long double t = res;
-        res *= std::numeric_limits<base_type>::max();
+        res *= static_cast<long double>(std::numeric_limits<base_type>::max());
```

Comment:
Not wrong but for education purposes it would be nicer to apply the workaround described in my other comment.